test/BatchExchangeViewer.spec.ts

```diff
@@ -531,4 +533,25 @@ contract("BatchExchangeViewer [ @skip-on-coverage ]", (accounts) => {
       assert.equal(decodeIndexedOrders(result.elements).length, 1);
     });
   });
+  describe("getTokenInfo", () => {
+    it("Allows to get token address, symbol and decimals by ID", async () => {
+      const erc20detailed = await ERC20Detailed.at(token_1.address);
+      const symbolMethod = erc20detailed.contract.methods.symbol().encodeABI();
+      const decimalsMethod = erc20detailed.contract.methods
+        .decimals()
+        .encodeABI();
+
+      await token_1.givenMethodReturn(
+        symbolMethod,
+        "0x" + abi.rawEncode(["string"], ["SCAM"]).toString("hex"),
+      );
+      await token_1.givenMethodReturnUint(decimalsMethod, 42);
+
+      const viewer = await BatchExchangeViewer.new(batchExchange.address);
+      const result = await viewer.getTokenInfo(1);
+      assert.equal(result[0], token_1.address);
+      assert.equal(result[1], "SCAM");
+      assert.equal(result[2].toNumber(), 42);
+    });
+  });
```

Comment:
There are tokens that do not implement `ERC20Detailed` (I'm looking at you `SAI`!). How does the SC act in those cases (I imagine it would just revert). Maybe include a unit test to assert this assumption.